### PR TITLE
Fixes the method used to detect stateset changed

### PR DESCRIPTION
**Pull request by stephomi**
The hash method was not robust enough

### CODE REVIEW

File: sources/osg/State.js [hunk 1]
```diff
@@ -1321,6 +1313,28 @@ State.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( Objec
             this._checkCacheAndApplyUniform( uniform, cacheUniformsForeign, i, programUniformMap, name );
 
         }
+    },
+
+    // Use to detect changed in RenderLeaf between call to avoid to applyStateSet
```
Comment (jtorresfabra) [inline sources/osg/State.js]: typo changed/changes

Comment (stephomi) [inline sources/osg/State.js]: yup

File: sources/osg/State.js [hunk 2]
```diff
@@ -1321,6 +1313,28 @@ State.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( Objec
...
+
+    // Use to detect changes in RenderLeaf between call to avoid to applyStateSet
+    _setStateSetsDrawID: function ( id ) {
+        var values = this.stateSets.values();
+        for ( var i = 0, nbStateSets = values.length; i < nbStateSets; i++ ) {
+            values[ i ].setDrawID( id );
+        }
+    },
+
+    _hasStateSetStackChangedFromPreviousDraw: function ( id, nbLast ) {
```
Comment (jtorresfabra) [inline sources/osg/State.js]: what do you think @stephomi?  _hasStateSetStackChangedFromPreviousDraw/_stateSetStackChangedFromPreviousDraw/ or simply  _stateSetStackChanged

Comment (stephomi) [inline sources/osg/State.js]: @cedricpinson changed the names

Comment (cedricpinson) [inline sources/osg/State.js]: I will update thx

**Pull request merged**